### PR TITLE
Display rejection transaction details

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #1269

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/res/drawable/ic_external_link_green_16dp.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="16dp"
+    android:height="16dp"
+    android:viewportWidth="16"
+    android:viewportHeight="16">
+  <path
+      android:pathData="M13,13v-2c0,-0.552 0.448,-1 1,-1s1,0.448 1,1v2c0,1.105 -0.895,2 -2,2H3c-1.105,0 -2,-0.895 -2,-2V3c0,-1.105 0.895,-2 2,-2h2c0.552,0 1,0.448 1,1s-0.448,1 -1,1H3v10h10z"
+      android:fillColor="#008C73"
```
Comment (biafra23) [inline app/src/main/res/drawable/ic_external_link_green_16dp.xml:8]: Can't we reuse `ic_link_green_24dp.xml`? If not, can you set the `fillColor` to `@color/primary` so it is correct in Dark mode?

File: app/src/main/res/drawable/ic_circle_cross_red_16dp.xml [hunk 2]
```diff
@@ -5,10 +5,10 @@
     android:viewportHeight="16">
   <path
       android:pathData="M8,15c3.866,0 7,-3.134 7,-7s-3.134,-7 -7,-7 -7,3.134 -7,7 3.134,7 7,7zM8,13c-2.761,0 -5,-2.239 -5,-5s2.239,-5 5,-5 5,2.239 5,5 -2.239,5 -5,5z"
-      android:fillColor="@color/error"
+      android:fillColor="#F02525"
```
Comment (biafra23) [inline app/src/main/res/drawable/ic_circle_cross_red_16dp.xml:8]: Why not `@color/error`?

File: app/src/main/res/layout/tx_details_rejection.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="@dimen/default_margin"
+        android:layout_marginLeft="@dimen/default_margin"
+        android:layout_marginRight="@dimen/default_margin"
+        android:layout_marginTop="10dp"
+        android:drawablePadding="6dp"
+        android:gravity="center_vertical"
+        android:text="@string/tx_details_rejection_payment_reason"
+        app:drawableEndCompat="@drawable/ic_external_link_green_16dp" />
```
Comment (biafra23) [inline app/src/main/res/layout/tx_details_rejection.xml:34]: This doesn't place the icon at the end of the text. When I had to create links with icons like this, I used `TextView.appendLink()` like in the following example:

```
sectionTitle.appendLink(
                url = resources.getString(R.string.tx_list_conflict_header_link),
                urlText = resources.getString(R.string.tx_list_conflict_header_learn_more),
                linkIcon = R.drawable.ic_link_green_24dp,
                prefix = " "
            )
```

File: app/src/main/res/layout/tx_details_rejection.xml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:animateLayoutChanges="true"
+    android:background="@color/surface_01"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/tx_rejection_info"
+        style="@style/TextDark.Bold"
```
Comment (biafra23) [inline app/src/main/res/layout/tx_details_rejection.xml:14]: IMHO This is not supposed to be bold. So `TextDark` would be the right style.

File: app/src/main/res/layout/tx_details_rejection.xml (hunk 3 above)
Comment (elgatovital) [reply app/src/main/res/layout/tx_details_rejection.xml:34]: we don't need it here. Whole text is clickable. This is how it looks
image

Comment (biafra23) [reply app/src/main/res/layout/tx_details_rejection.xml]: On my phone it looked like this: 
image

**Review by biafra23 — APPROVED** [review]

File: app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt [hunk 5]
```diff
@@ -371,6 +371,7 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
                 val txDetailsRejectionBinding = contentBinding as TxDetailsRejectionBinding
                 with(txDetailsRejectionBinding) {
                     txRejectionInfo.text = getString(R.string.tx_details_rejection_info, nonce)
+                    txPaymentReasonLink.text = null
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt:374]: Sorry. I should have warned you about this append behaviour. Should we have a warning about this as a comment?

Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt]: check out new set link extension

File: app/src/main/java/io/gnosis/safe/utils/StringUtils.kt [hunk 6]
```diff
@@ -158,3 +158,15 @@ fun TextView.appendLink(
...
+
+fun TextView.setLink(
+    url: String,
+    urlText: String,
+    @DrawableRes linkIcon: Int? = null,
+    @ColorRes textColor: Int = R.color.primary,
+    prefix: String = "",
+    underline: Boolean = false
+) {
+    text = null
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/utils/StringUtils.kt:170]: Nice! 🚀

**Pull request merged**